hack/test-cmd.sh

```diff
@@ -239,84 +239,6 @@ os::test::junit::declare_suite_start "cmd/config"
 # ensure that DisabledFeatures aren't written to config files
 os::cmd::expect_success_and_text "cat ${MASTER_CONFIG_DIR}/master-config.yaml" 'disabledFeatures: null'
 os::cmd::expect_success_and_text "cat ${BASETMPDIR}/atomic.local.config/master/master-config.yaml" 'disabledFeatures: null'
-
-# test client not configured
-os::cmd::expect_failure_and_text "oc get services" 'Missing or incomplete configuration info.  Please login'
-unused_port="33333"
-# setting env bypasses the not configured message
-os::cmd::expect_failure_and_text "KUBERNETES_MASTER=http://${API_HOST}:${unused_port} oc get services" 'did you specify the right host or port'
-# setting --server bypasses the not configured message
-os::cmd::expect_failure_and_text "oc get services --server=http://${API_HOST}:${unused_port}" 'did you specify the right host or port'
-
-# Set KUBERNETES_MASTER for oc from now on
-export KUBERNETES_MASTER="${API_SCHEME}://${API_HOST}:${API_PORT}"
-
-# Set certificates for oc from now on
-if [[ "${API_SCHEME}" == "https" ]]; then
-    # test bad certificate
-    os::cmd::expect_failure_and_text "oc get services" 'certificate signed by unknown authority'
-fi
-
-# login and logout tests
```

Comment:
> making the login test undo the setup the main script does seems weird.

I disagree. We want a test that stands alone and does not make assumptions about what has and has not run before it. Furthermore, we want the launch script for `test-cmd.sh` to launch the server and run the tests the user asked for, not _also_ 200 tests that we happen to have there at the moment.

By making an encapsulated test for login, we get all those benefits. If our tests don't fail when we _are_ logged in ... those are some bad tests.

> are you certain you're unsetting all the envvars to get back to a completely clean state

About 95% certain, the other 5% I was hoping to get feedback from you about.

> (KUBERNETES_MASTER springs to mind)

From the login test:
```sh
# we want this test to run without $KUBECONFIG or $KUBERNETES_MASTER as it tests that functionality
unset KUBECONFIG
unset KUBERNETES_MASTER
```